### PR TITLE
Downgrade CI

### PR DESCRIPTION
**Pull request by ArnoStrouwen**

### CODE REVIEW

**Comment by ArnoStrouwen** [comment]
Regression?

**Comment by ChrisRackauckas** [comment]
```julia
for i = 1:n
        #= /home/runner/work/MuladdMacro.jl/MuladdMacro.jl/test/runtests.jl:104 =#
        z = (muladd)(x, i, y)
        #= /home/runner/work/MuladdMacro.jl/MuladdMacro.jl/test/runtests.jl:105 =#
    end == for i = 1:n
        #= /home/runner/work/MuladdMacro.jl/MuladdMacro.jl/test/runtests.jl:107 =#
        z = (muladd)(x, i, y)
        #= /home/runner/work/MuladdMacro.jl/MuladdMacro.jl/test/runtests.jl:108 =#
    end
```

Is the regression in `==` on Expr?

**Comment by ArnoStrouwen** [comment]
```jl
julia>     @test @macroexpand(@muladd f(x, y, z) = x * y + z) ==
                 :(f(x, y, z) = $(Base.muladd)(x, y, z))
Test Failed at REPL[29]:1
  Expression: #= REPL[29]:1 =# @macroexpand(#= REPL[29]:1 =# @muladd(f(x, y, z) = begin
                    #= REPL[29]:1 =#
                    x * y + z
                end)) == :(f(x, y, z) = begin
              #= REPL[29]:2 =#
              ($(Base.muladd))(x, y, z)
          end)
   Evaluated: (f(x, y, z) = begin
            #= REPL[29]:1 =#
            (muladd)(x, y, z)
        end) == (f(x, y, z) = begin
            #= REPL[29]:2 =#
            (muladd)(x, y, z)
        end)
```

**Comment by ChrisRackauckas** [comment]
It was pointed out on Slack that it may be the line numbers? We can try stipping the line numbers before the comparison.

**Comment by ArnoStrouwen** [comment]
Good to go now.

File: Project.toml [hunk 1]
```diff
@@ -3,9 +3,9 @@ uuid = "46d2c3a1-f734-5fdb-9937-b9b9aeba4221"
 version = "0.2.4"
 
 [compat]
-Aqua = "0.8"
-Test = "1"
-julia = "1.6"
+Aqua = "0.8.4"
+Test = "1.0.0"
+julia = "1.10"
```
Comment (devmotion) [inline Project.toml:8]: This change seems unnecessary?

Comment (ArnoStrouwen) [reply Project.toml:8]: @ChrisRackauckas decided to move everything to Julia 1.10 as it is likely to be the new LTS.

Comment (devmotion) [reply Project.toml:8]: I think such unnecessary Julia compat changes are quite unfortunate for utility packages such as MuladdMacro that are useful not only for SciML packages but the whole ecosystem.

Comment (ChrisRackauckas) [reply Project.toml:8]: It's somewhat easiest to just have everything on what will be the new LTS though. At least as we're rolling out the new Downgrade CI which is a surprisingly difficult thing to get right, keeping everything simple is best. I do agree that we could in theory make exceptions for something like this repo which effectively never changes though (though its tests actually needed a change for v1.10)

**Pull request merged**